### ISSUE TITLE
Diagram plots not showing up

### ISSUE THREAD

**Issue by matthewmacleod** [ricklupton/floweaver#52]
First off, thanks for the cool library. I am having some problems getting this to work on Ubuntu 16.04.  When I execute the quickstart notebook (after following installation directions and installation of the requirements.txt libraries), the diagrams do not visually display, instead only some text is displayed (see screenshot below.)

![screenshot from 2018-07-09 09-30-01](https://user-images.githubusercontent.com/2094725/42460312-c7278712-835a-11e8-8983-a3a139c9d306.png)

I'm not sure what the problem is and have tried various virtualenv and conda setups. The current configuration is:

python --version
Python 3.6.2 :: Anaconda, Inc.

pip freeze
alabaster==0.7.11
attrs==18.1.0
Babel==2.6.0
backcall==0.1.0
bleach==2.1.3
certifi==2018.4.16
chardet==3.0.4
cycler==0.10.0
decorator==4.3.0
defusedxml==0.5.0
docutils==0.14
entrypoints==0.2.3
floweaver==2.0.0a3
html5lib==1.0.1
idna==2.7
imagesize==1.0.0
ipykernel==4.8.2
ipysankeywidget==0.2.4
ipython==6.4.0
ipython-genutils==0.2.0
ipywidgets==7.2.1
jedi==0.12.1
Jinja2==2.10
jsonschema==2.6.0
jupyter-client==5.2.3
jupyter-core==4.4.0
kiwisolver==1.0.1
MarkupSafe==1.0
matplotlib==2.2.2
mistune==0.8.3
nbconvert==5.3.2.dev0
nbformat==4.4.0
nbsphinx==0.3.3
networkx==1.11
notebook==5.5.0
numpy==1.14.5
packaging==17.1
palettable==3.1.1
pandas==0.23.3
pandocfilters==1.4.2
parso==0.3.0
pexpect==4.6.0
pickleshare==0.7.4
prompt-toolkit==1.0.15
ptyprocess==0.6.0
Pygments==2.2.0
pyparsing==2.2.0
python-dateutil==2.7.3
pytz==2018.5
pyzmq==17.0.0
requests==2.19.1
Send2Trash==1.5.0
simplegeneric==0.8.1
six==1.11.0
snowballstemmer==1.2.1
Sphinx==1.7.5
sphinxcontrib-websupport==1.1.0
terminado==0.8.1
testpath==0.3.1
tornado==5.0.2
traitlets==4.3.2
urllib3==1.23
wcwidth==0.1.7
webencodings==0.5.1
widgetsnbextension==3.2.1

I am tried chrome (Version 65.0.3325.181 (Official Build) (64-bit)) and firefox (60.0.2 64-bit). I'm using Ubuntu 16.04.4 LTS.

**Comment by matthewmacleod** [comment]
It seems the issue is not related to this library but just getting the widgets to work. Maybe some additional instructions would be helpful.  I fixed this issue by doing this:

jupyter nbextension install --py widgetsnbextension --user

And then I did the activation step

jupyter nbextension enable widgetsnbextension --user --py

Cheers

**Comment by ricklupton** [comment]
Hi @matthewmacleod, glad you got it working. This step is mentioned in the [Getting Started section](https://github.com/ricklupton/floweaver#get-started-using-floweaver) of the README, but I'm saying that to figure out if it could be clearer rather than to blame you for having missed it! Suggestions / pull requests for changes to the README and documentation welcome to make this more clearer.

**Comment by ralienpp** [comment]
Unfortunately the provided instructions do not work, here is everything I am doing:

1. `pip install --user floweaver ipysankeywidget`
2. `jupyter nbextension install --py widgetsnbextension --user`
3. `jupyter nbextension enable widgetsnbextension --user --py`

Note that I have the `--user` switch, because this is running in an environment where I have no permission to make global changes.

I get no error messages, unlike the original poster, Jupyter just moves on to the next cell.

![image](https://user-images.githubusercontent.com/295338/48488717-b1d6a880-e821-11e8-8c6d-aa58c74332a8.png)


Here are the package versions I have at the moment:

```
alembic==0.9.9
altair==2.0.0rc1appdirs==1.4.3
asn1crypto==0.24.0async-generator==1.10
attrs==17.4.0Automat==0.0.0backcall==0.1.0
beautifulsoup4==4.6.1bleach==2.1.3bokeh==0.12.16
Bottleneck==1.2.1
certifi==2018.8.24
cffi==1.11.5
cftime==1.0.1
chardet==3.0.4
click==6.7
cloudpickle==0.5.3
colorcet==1.0.0conda==4.5.11
constantly==15.1.0
cryptography==2.2.1
cycler==0.10.0
Cython==0.28.5
cytoolz==0.9.0.1
dask==0.18.2
datashader==0.6.6
datashape==0.5.4
decorator==4.3.0
dill==0.2.8.2
distributed==1.23.1
entrypoints==0.2.3
fastcache==1.0.2
floweaver==2.0.0a4
gmpy2==2.0.8
h5netcdf==0.6.2
h5py==2.7.1
heapdict==1.0.0
html5lib==1.0.1
hyperlink==17.3.1
idna==2.7
imageio==2.3.0
incremental==17.5.0
ipykernel==4.8.2
ipysankeywidget==0.2.4
ipython==6.5.0
ipython-genutils==0.2.0
ipywidgets==7.2.1
jedi==0.12.1
Jinja2==2.10
jsonschema==2.6.0
jupyter-client==5.2.3
jupyter-core==4.4.0
jupyterhub==0.9.1
jupyterlab==0.33.7
jupyterlab-launcher==0.11.2
kiwisolver==1.0.1
llvmlite==0.23.0
locket==0.2.0
Mako==1.0.7
MarkupSafe==1.0
matplotlib==2.2.2
mistune==0.8.3
msgpack==0.5.6
multipledispatch==0.6.0
nbconvert==5.3.1
nbformat==4.4.0
netCDF4==1.4.0
networkx==1.11
notebook==5.6.0
numba==0.38.1
numexpr==2.6.6
numpy==1.13.3
olefile==0.45.1
packaging==17.1
palettable==3.1.1
pamela==0.3.0
pandas==0.23.4
pandocfilters==1.4.2
param==1.7.0
parso==0.3.1
partd==0.3.8
patsy==0.5.0
pdvega==0.1
pexpect==4.6.0
pickleshare==0.7.4
Pillow==5.2.0
pluggy==0.6.0
prometheus-client==0.3.0
prompt-toolkit==1.0.15
protobuf==3.5.2
psutil==5.4.7
ptyprocess==0.6.0
py==1.5.2
pyasn1==0.4.4
pyasn1-modules==0.2.1
pycosat==0.6.3
pycparser==2.18
pyct==0.4.3
pycurl==7.43.0.2
Pygments==2.2.0
pyOpenSSL==18.0.0
pyparsing==2.2.0
PySocks==1.6.8
pytest==3.4.2
python-dateutil==2.7.3
python-editor==1.0.3
python-oauth2==1.0.1
pytz==2018.5
PyWavelets==0.5.2
PyYAML==3.12
pyzmq==17.1.0
requests==2.19.1
rpy2==2.8.5
ruamel-yaml==0.15.44
scikit-image==0.14.0
scikit-learn==0.19.2
scipy==1.1.0
seaborn==0.9.0
Send2Trash==1.5.0
service-identity==17.0.0
simplegeneric==0.8.1
singledispatch==3.4.0.3
six==1.11.0
sortedcontainers==2.0.4
SQLAlchemy==1.2.10
statsmodels==0.9.0
sympy==1.1.1
tblib==1.3.2
terminado==0.8.1
testpath==0.3.1
toolz==0.9.0
tornado==5.1
traitlets==4.3.2
Twisted==18.7.0
urllib3==1.23
vega==0.4.4
vega-datasets==0.4.1
vega3==0.4.0
vincent==0.4.4
wcwidth==0.1.7
webencodings==0.5
widgetsnbextension==3.2.1
xarray==0.10.8
xlrd==1.1.0
zict==0.1.3
zope.interface==4.5.0
```

The funny thing is that I've had the same thing happen to me yesterday, but after fiddling around for a while, I managed to get the expected result. I wrote down the instructions above, for my future self. The problem is that they are not sufficient for a clean start, there must have been something else that played a role, because now I cannot replicate the result.

Note that I widgets work, in general. You can see that Pandas' dataframe is rendered correctly:

![image](https://user-images.githubusercontent.com/295338/48489111-915b1e00-e822-11e8-8be7-d8c0b6307abb.png)

What other troubleshooting steps can I try?

**Comment by ricklupton** [comment]
Hi @ralienpp the other step listed in the [installation instructions](https://github.com/ricklupton/floweaver#get-started-using-floweaver) is to activate ipysankeywidget nbextension, have you done that too?

```shell
jupyter nbextension enable --py --user ipysankeywidget
```

**Comment by ralienpp** [comment]
Thank you for the swift reply. Now I think I understand why it worked yesterday - because I followed the installation instructions first, pause, then I applied the steps described by Matthew, pause, then I documented them.

Here is a minor tweak to the doc, that could make things a bit smoother: https://github.com/ricklupton/floweaver/pull/66

Another remark worth adding is that if the problem persists, one should close the kernel and start a new session. I didn't add this to the doc, perhaps this could go into a separate _"Troubleshooting"_ section, if other materials for it exist.
